commit cfa2e1e5c24f3ae16412c55aaef2ffc6197a0465 (refs/changes/29/3729/1)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-03-27T16:21:59-04:00 (1 year, 6 months ago)
    
    TRITON-280 vmapi vms.full.test should block on test VMs to be destroyed fully

diff --git a/test/vms.full.test.js b/test/vms.full.test.js
index 04fcf87..e722f50 100644
--- a/test/vms.full.test.js
+++ b/test/vms.full.test.js
@@ -184,6 +184,9 @@ function createTestVms(cb) {
                     return;
                 }
 
+                assert.object(body, 'body');
+                assert.uuid(body.job_uuid, 'body.job_uuid');
+
                 var job = '/jobs/' + body.job_uuid;
 
                 i++;
@@ -241,8 +244,15 @@ function destroyTestVms(cb) {
                         return;
                     }
 
+                    assert.object(delBody, 'delBody');
+                    assert.uuid(delBody.job_uuid, 'delBody.job_uuid');
+
+                    var job = '/jobs/' + delBody.job_uuid;
+
                     ret.destroyedVms++;
-                    next();
+                    waitForValue(job, 'execution', 'succeeded', {
+                        client: client
+                    }, next);
                 });
 
             }, function (delVmsErr) {
